x-pack/dockerlogbeat/pipelineManager/pipelineManager.go

```diff
@@ -0,0 +1,233 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package pipelinemanager
+
+import (
+	"crypto/sha1"
+	"fmt"
+	"sort"
+	"sync"
+
+	"github.com/elastic/beats/libbeat/beat"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/idxmgmt"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/libbeat/outputs"
+	"github.com/elastic/beats/libbeat/publisher/pipeline"
+	"github.com/elastic/beats/libbeat/publisher/processing"
+	"github.com/pkg/errors"
+	yaml "gopkg.in/yaml.v2"
+)
+
+// containerConfig is the common.Config unpacking type
+type containerConfig struct {
+	Pipeline pipeline.Config        `config:"pipeline"`
+	Output   common.ConfigNamespace `config:"output"`
+}
+
+// Pipeline represents a single pipeline and the count of associated clients
+type Pipeline struct {
+	pipeline *pipeline.Pipeline
+	refCount int
+}
+
+// PipelineManager is a handler into the map of pipelines used by the plugin
+type PipelineManager struct {
+	mu     *sync.Mutex
```

Comment:
why `*sync.Mutex`? Do you want `PipelineManager` as is to be shallow-copyable?

Reply:
Honestly not sure what my reasoning was there. Should probably remove it.